### PR TITLE
feat: update default env images to pytorch 1.10.2, tf 2.8.

### PR DESCRIPTION
**Pull request by ioga**
## Description

- horovod 0.24.2.
- minor updates to cifar examples related to tf 2.8
- some updates to mmcv and mmdetection.
- add legacy tf24-{cpu,gpu} images.

## Test Plan

Run your favorite tricky dtrain experiment.  

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by gh-determined-ai — APPROVED** [review]
What's also needed or, maybe, instead of, is a single-page reference that lists the current versions. It's a little crazy having these spread throughout the documentation. I'll add that to my list of things to do.

You should assign someone else to approve the non-doc files that changed.

File: model_hub/model_hub/mmdetection/_callbacks.py [hunk 1]
```diff
@@ -141,8 +141,9 @@ def __init__(self, context: det_torch.PyTorchTrialContext, model: torch.nn.Modul
 
     def on_validation_start(self) -> None:
         if self.context.distributed.get_size() > 1:
-            logging.info("Broadcasting batch norm stats from chief for eval.")
+            logging.info("Broadcasting batch norm stats from chief for eval - HALLO.")
```
Comment (mackrorysd) [inline model_hub/model_hub/mmdetection/_callbacks.py:144]: Did you mean for this to get left in?

File: model_hub/Makefile [hunk 2]
```diff
@@ -17,10 +17,10 @@ TRANSFORMERS_VERSION := 4.8.2
 DATASETS_VERSION := 1.9.0
 TRANSFORMERS_ENVIRONMENT_ROOT := determinedai/model-hub-transformers
 # MMDetection Args
-MMCV_CUDA_VERSION := cu111 # Needs to match DEFAULT_GPU_IMAGE
-TORCH_VERSION := 1.9.0 # Needs to match DEFAULT_GPU_IMAGE
-MMCV_VERSION := 1.3.12
-MMDETECTION_VERSION := 2.16.0
+MMCV_CUDA_VERSION := cu113 # Needs to match DEFAULT_GPU_IMAGE
```
Comment (mackrorysd) [inline model_hub/Makefile:20]: This'll easily get forgotten. We should add a blurb about this to the bumpenvs README if, as I assume, this doesn't get updated automatically. Though I guess tests fail if you don't do it when you need it?

Comment (ioga) [reply model_hub/Makefile:20]: yes, tests will fail. 

this is hardly a thing for bumpenvs - just like a support for new TF versions. I am not sure what kind of a README will be helpful for this.

File: model_hub/model_hub/mmdetection/_callbacks.py (hunk 1 above)
Comment (ioga) [reply model_hub/model_hub/mmdetection/_callbacks.py]: fixed

File: model_hub/Makefile (hunk 2 above)
Comment (mackrorysd) [reply model_hub/Makefile:20]: Just thinking that bumping CUDA versions and TORCH versions happens in a non-trivial portion of bumpenvs, even if not the majority. But as long as the tests do catch it, fair enough.

**Review by mackrorysd — APPROVED** [review]
